### PR TITLE
RFX

### PR DESCRIPTION
**Pull request by ndavison**

### CODE REVIEW

File: apps/marketplace/components/Opportunity/OpportunityInfoCard.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            {format(props.closingDate, 'D MMMM YYYY')}
+          </p>
+        )}
+        {!props.isClosed &&
+          props.isInvitedSeller && (
+            <div>
+              {props.hasResponded ? (
+                <p className={styles.invitedStatus}>You have already applied for this opportunity.</p>
+              ) : (
+                <a href={`${rootPath}/brief/${props.briefId}/rfx/respond`} className={`${styles.button} au-btn`}>
```
Comment (malmas18) [inline apps/marketplace/components/Opportunity/OpportunityInfoCard.js]: can this be made generic?

File: apps/marketplace/MarketplaceAuthenticatedRoute.js [hunk 2]
```diff
@@ -37,18 +50,21 @@ const PrivateRouteComponent = props => {
...
+  restrictedTo: null
 }
 
 PrivateRouteComponent.propTypes = {
   component: PropTypes.func.isRequired,
   loggedIn: PropTypes.bool.isRequired,
   customRedirectPath: PropTypes.string,
-  currentlySending: PropTypes.bool
+  currentlySending: PropTypes.bool,
+  restrictedTo: PropTypes.oneOf(['buyer', 'supplier'])
```
Comment (malmas18) [inline apps/marketplace/MarketplaceAuthenticatedRoute.js]: might need to add `admin` here

File: apps/marketplace/components/BuyerRFX/BuyerRFXEvaluationCriteriaStage.js [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+class BuyerRFXEvaluationCriteriaStage extends Component {
+  constructor(props) {
+    super(props)
+    this.handleIncludeWeightingsChange = this.handleIncludeWeightingsChange.bind(this)
+    this.handleAddCriteriaClick = this.handleAddCriteriaClick.bind(this)
+    this.removeCriteria = this.removeCriteria.bind(this)
+  }
+
+  getRemainingWeighting() {
```
Comment (tobiasbrown) [inline apps/marketplace/components/BuyerRFX/BuyerRFXEvaluationCriteriaStage.js]: This needs some extra checks on the input.  I stumbled across a negative number.
screen shot 2018-12-15 at 12 26 05 pm

Comment (ndavison) [reply apps/marketplace/components/BuyerRFX/BuyerRFXEvaluationCriteriaStage.js]: This is by design.

File: apps/marketplace/components/BuyerRFX/BuyerRFXEvaluationCriteriaStage.js [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+    return remaining
+  }
+
+  removeCriteria(index) {
+    this.props.removeCriteriaByIndex(this.props[this.props.model].evaluationCriteria, index)
+  }
+
+  handleAddCriteriaClick(e) {
+    e.preventDefault()
+    this.props.addEmptyEvalutationCriteria(this.props[this.props.model].evaluationCriteria)
```
Comment (tobiasbrown) [inline apps/marketplace/components/BuyerRFX/BuyerRFXEvaluationCriteriaStage.js]:
```suggestion
    this.props.addEmptyEvaluationCriteria(this.props[this.props.model].evaluationCriteria)
```

File: apps/marketplace/components/BuyerRFX/BuyerRFXEvaluationCriteriaStage.js [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+        </div>
+        {this.props.formButtons}
+      </Form>
+    )
+  }
+}
+
+BuyerRFXEvaluationCriteriaStage.defaultProps = {
+  clearWeightingsFromCriteria: () => {},
+  addEmptyEvalutationCriteria: () => {},
```
Comment (tobiasbrown) [inline apps/marketplace/components/BuyerRFX/BuyerRFXEvaluationCriteriaStage.js]:
```suggestion
  addEmptyEvaluationCriteria: () => {},
```

File: apps/marketplace/components/BuyerRFX/BuyerRFXEvaluationCriteriaStage.js [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+  removeCriteriaByIndex: () => {},
+  onSubmit: () => {},
+  onSubmitFailed: () => {}
+}
+
+BuyerRFXEvaluationCriteriaStage.propTypes = {
+  model: PropTypes.string.isRequired,
+  clearWeightingsFromCriteria: PropTypes.func,
+  addEmptyEvalutationCriteria: PropTypes.func,
```
Comment (tobiasbrown) [inline apps/marketplace/components/BuyerRFX/BuyerRFXEvaluationCriteriaStage.js]:
```suggestion
  addEmptyEvaluationCriteria: PropTypes.func,
```

File: apps/marketplace/components/BuyerRFX/BuyerRFXEvaluationCriteriaStage.js [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+const mapDispatchToProps = (dispatch, props) => ({
+  clearWeightingsFromCriteria: evaluationCriteria => {
+    const newData = evaluationCriteria.map(evaluation => {
+      const newEvaluation = { ...evaluation }
+      newEvaluation.weighting = ''
+      return newEvaluation
+    })
+    dispatch(actions.change(`${props.model}.evaluationCriteria`, newData))
+  },
+  addEmptyEvalutationCriteria: evaluationCriteria => {
```
Comment (tobiasbrown) [inline apps/marketplace/components/BuyerRFX/BuyerRFXEvaluationCriteriaStage.js]:
```suggestion
  addEmptyEvaluationCriteria: evaluationCriteria => {
```

File: apps/marketplace/components/NoticeBar/NoticeBar.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styles from './NoticeBar.scss'
+
+const NoticeBar = props => (
```
[the remark this answers is not in the document]
Comment (tobiasbrown) [reply apps/marketplace/components/NoticeBar/NoticeBar.js]: @alex-page / @sukhrajghuman / @adamzerella, you might be interested in these test results.

[the remark this answers is not in the document]
Comment (ndavison) [reply apps/marketplace/components/NoticeBar/NoticeBar.js]: It's actually pretty similar to the callout component, but with a thicker left border and different colouring. @tobiasbrown we actually already have callout in our package.json so I might try and use that instead with different styles applied.

[the remark this answers is not in the document]
Comment (alex-page) [reply apps/marketplace/components/NoticeBar/NoticeBar.js]: Feel free to use `au-callout` maybe add a new class called `au-callout--warning` `au-callout--error` and add the styles you need.

**Pull request merged**